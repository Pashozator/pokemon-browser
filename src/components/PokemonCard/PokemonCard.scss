@import '../../styles/mixins';
@import '../../styles/media-queries';

.pokemon-card {
  margin: 0 0 10px 0;
  padding: 25px;
  width: 100%;
  height: auto;

  @include for-up-sm() {
    margin-right: 10px;
    width: 400px;
    height: 400px;
  }

  @include hover() {
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, .3);
  }

  img {
    width: 100%;
    height: 100%;
  }
}
