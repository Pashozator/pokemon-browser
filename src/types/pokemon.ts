export interface Pokemon {
	id: string;
	number: number;
	name: string;
	image: string;
	maxHP: number;
	fleeRate: number;
	maxCP: number;
	classification: string;
	types: string[];
	resistant: string[];
	weaknesses: string[];
	evolutions: Pokemon[];
}
