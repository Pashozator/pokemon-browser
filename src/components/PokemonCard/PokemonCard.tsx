import React from 'react';
import './PokemonCard.scss';
import { Pokemon } from '../../types/pokemon';
import { Link } from 'react-router-dom';
import Card from '../Card/Card';

interface PokemonCardProps {
	pokemon: Pokemon;
}

const PokemonCard: React.FC<PokemonCardProps> = ({ pokemon }) => {
	return (
		<Link to={`/${pokemon.name}`}>
			<button>
				<Card className="pokemon-card">
					{pokemon.name}
					<img src={pokemon.image} alt={pokemon.name} />
				</Card>
			</button>
		</Link>
	)
};

export default PokemonCard;
