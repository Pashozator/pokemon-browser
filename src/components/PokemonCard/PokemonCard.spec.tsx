import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PokemonCard from './PokemonCard';
import { Pokemon } from '../../types/pokemon';

describe('PokemonCard component', () => {
	it('should render', () => {
		const pokemon: Pokemon = {
			id: 'test',
			name: 'Pikachu',
			image: 'image'
		};

		const wrapper = shallow(<PokemonCard pokemon={pokemon} />);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
