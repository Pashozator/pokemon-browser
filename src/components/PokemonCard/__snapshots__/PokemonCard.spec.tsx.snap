// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard component should render 1`] = `
<Link
  to="/Pikachu"
>
  <button>
    <Card
      className="pokemon-card"
    >
      Pikachu
      <img
        alt="Pikachu"
        src="image"
      />
    </Card>
  </button>
</Link>
`;
